src/client/components/scatter_plot/view.tsx

```diff
@@ -1,5 +1,256 @@
+    }
+
+    // if we are currently updating the plotly trace
+    if (trace.display) {
+      let updateX = this.graphUpdateX.get(trace.id)
+      let updateY = this.graphUpdateY.get(trace.id)
+      let updateZ = this.graphUpdateZ.get(trace.id)
+
+      // if no updates to the trace has been made since the last graph update
+      // add it to the update lists
+      if (updateX == null) {
+        updateX = []
+        this.graphUpdateX.set(trace.id, updateX)
+      }
+      if (updateY == null) {
+        updateY = []
+        this.graphUpdateY.set(trace.id, updateY)
+      }
+      if (updateZ == null) {
+        updateZ = []
+        this.graphUpdateZ.set(trace.id, updateZ)
+      }
+
+      // append our new data to the update lists
+      if (data.values.length === 1) {
+        trace.xVal += 1
+        updateX.push(trace.xVal)
+        updateY.push(data.values[0])
+        updateZ.push(0)
+      } else if (data.values.length === 2) {
+        updateX.push(data.values[0])
+        updateY.push(data.values[1])
+        updateZ.push(0)
+      } else if (data.values.length >= 3) { // TODO: work out how to deal with data points with more then 3 values
+        updateX.push(data.values[0])
+        updateY.push(data.values[1])
+        updateZ.push(data.values[2])
+      }
+    }
+  }
+
+  public addTrace(label: string): void {
+    const newTrace: Plotly.ShortData = {
+      x: [],
+      y: [],
+      z: [],
+      mode: 'markers',
+      type: 'scattergl',
+      marker: {
+        size: 10,
+      },
+      name: label,
+    }
+
+    Plotly.addTraces(this.canvas, newTrace)
+  }
+
+  public createTrace(label: string): Trace {
+    const trace: Trace = {
+      mode: 'markers',
+      type: 'scattergl',
+      hoverinfo: 'x+y',
+      marker: {size: 12},
+      name: label,
+      xVal: 0,
+      id: -1,
+      display: true, // TODO: set this to false, and create a config panel to enable it
+      addTrace: true, // TODO: set this to false, and create a config panel to enable it
+    }
+    return trace
+  }
+
+  public updateDimensions(): void {
+    const layout = {
+      width: this.canvas.offsetWidth,
+      height: this.canvas.offsetHeight,
+    }
+    Plotly.relayout(this.canvas, layout)
+  }
+
+  public componentDidMount(): void {
+    window.addEventListener('resize', this.updateDimensions.bind(this))
+
+    const data: any = [
+    ]
+
+    Plotly.newPlot(this.canvas, data)
+
+    const me = this
+
+    this.updateLoopId = window.setInterval(() => {
+      const update = {
+        x: Array.from(me.graphUpdateX.values()),
+        y: Array.from(me.graphUpdateY.values()),
+        z: Array.from(me.graphUpdateZ.values()),
+      }
+
+      Plotly.extendTraces(this.canvas, update, Array.from(me.graphUpdateX.keys()), me.maxPoints)
+
+      me.graphUpdateX.clear()
+      me.graphUpdateY.clear()
+      me.graphUpdateZ.clear()
+    }, 1000 / this.fps)
+
+    this.simulateData()
+  }
+
+  public simulateData(): void {
```

Comment:
I am not too sure about smoothie, I think it is really only label and data.

Plotly how ever requires x, y, z, marker info (size, colour, icon etc), type (scatterplot, line graph, contour, histogram etc), hover info, label, id (the id that plotly assigns the trace so we can add data to it), and most likely even more.

The plotly graph it self has some specific stuff as well, such as controlling the axis min and max if we want to be able to restrict the graph to a certain area.